### PR TITLE
boards/common/blxxxpill: DMA feature added

### PR DESCRIPTION
**Pull request by hugueslarrive**
### Contribution description
Added DMA feature to check that dma_stop in spi.c isn't only needed by F3 family for  #14764.

### Testing procedure
Running tests/periph_spi_dma.

### Issues/PRs references
Forked from PR #14764.

### CODE REVIEW

File: cpu/stm32/periph/dma.c [hunk 1]
```diff
@@ -196,6 +192,7 @@ static IRQn_Type dma_get_irqn(int stream)
     if (stream < 7) {
         return ((IRQn_Type)((int)DMA1_Channel1_IRQn + stream));
     }
+#if !defined(CPU_FAM_STM32F3) && !defined(CPU_LINE_STM32F103xB)
```
Comment (benpicco) [inline cpu/stm32/periph/dma.c:195]: That seems pretty fragile
Can't we do

```C
#ifdef DMA2_BASE
…
```

**Comment by fjmolinas** [comment]
This one needs a rebase now.

**Comment by hugueslarrive** [comment]
```
2020-08-19 14:16:17,985 #  reboot
2020-08-19 14:16:17,991 # main(): This is RIOT! (Version: 2020.10-devel-845-gc7836-boards/common/blxxxpill)
2020-08-19 14:16:17,994 # Manual SPI peripheral driver test
2020-08-19 14:16:17,999 # Refer to the README.md file for more information.
2020-08-19 14:16:17,999 # 
2020-08-19 14:16:18,004 # There are 2 SPI devices configured for your platform.
init 0 0 4 0 0
2020-08-19 14:16:26,298 #  init 0 0 4 0 0
2020-08-19 14:16:26,304 # SPI_DEV(0) initialized: mode: 0, clk: 4, cs_port: 0, cs_pin: 0
bench
2020-08-19 14:16:32,990 #  bench
2020-08-19 14:16:32,993 # ### Running some benchmarks, all values in [us] ###
2020-08-19 14:16:32,996 # ### Test				Transfer time	user time
2020-08-19 14:16:32,997 # 
2020-08-19 14:16:33,015 #  1 - write 1000 times 1 byte:			13418	13423
2020-08-19 14:16:33,032 #  2 - write 1000 times 2 byte:			13231	13236
2020-08-19 14:16:33,142 #  3 - write 1000 times 100 byte:		105863	15374
2020-08-19 14:16:33,173 #  4 - write 1000 times 1 byte to register:	26384	26388
2020-08-19 14:16:33,205 #  5 - write 1000 times 2 byte to register:	26356	26361
2020-08-19 14:16:33,329 #  6 - write 1000 times 100 byte to register:	118946	28448
2020-08-19 14:16:33,345 #  7 - read 1000 times 2 byte:			13169	13172
2020-08-19 14:16:33,455 #  8 - read 1000 times 100 byte:		105649	15150
2020-08-19 14:16:33,487 #  9 - read 1000 times 2 byte from register:	26237	26242
2020-08-19 14:16:33,611 # 10 - read 1000 times 100 byte from register:	118842	28347
2020-08-19 14:16:33,628 # 11 - transfer 1000 times 2 byte:		13036	13041
2020-08-19 14:16:33,738 # 12 - transfer 1000 times 100 byte:		105647	15152
2020-08-19 14:16:33,769 # 13 - transfer 1000 times 2 byte to register:	26217	26222
2020-08-19 14:16:33,893 # 14 - transfer 1000 times 100 byte to register:118814	28319
2020-08-19 14:16:33,915 # 15 - acquire/release 1000 times:		17703	17708
2020-08-19 14:16:34,916 # -- - SUM:					849512	306583
2020-08-19 14:16:34,917 # 
2020-08-19 14:16:34,920 # ### All runs complete ###
init 1 0 4 0 0
2020-08-19 14:17:36,817 #  init 1 0 4 0 0
2020-08-19 14:17:36,824 # SPI_DEV(1) initialized: mode: 0, clk: 4, cs_port: 0, cs_pin: 0
bench
2020-08-19 14:17:48,834 #  bench
2020-08-19 14:17:48,837 # ### Running some benchmarks, all values in [us] ###
2020-08-19 14:17:48,840 # ### Test				Transfer time	user time
2020-08-19 14:17:48,840 # 
2020-08-19 14:17:48,859 #  1 - write 1000 times 1 byte:			13557	13561
2020-08-19 14:17:48,876 #  2 - write 1000 times 2 byte:			13231	13235
2020-08-19 14:17:48,986 #  3 - write 1000 times 100 byte:		106076	15533
2020-08-19 14:17:49,018 #  4 - write 1000 times 1 byte to register:	26648	26652
2020-08-19 14:17:49,049 #  5 - write 1000 times 2 byte to register:	26509	26514
2020-08-19 14:17:49,173 #  6 - write 1000 times 100 byte to register:	119287	28737
2020-08-19 14:17:49,190 #  7 - read 1000 times 2 byte:			13169	13172
2020-08-19 14:17:49,300 #  8 - read 1000 times 100 byte:		105926	15347
2020-08-19 14:17:49,332 #  9 - read 1000 times 2 byte from register:	26390	26395
2020-08-19 14:17:49,456 # 10 - read 1000 times 100 byte from register:	119198	28675
2020-08-19 14:17:49,473 # 11 - transfer 1000 times 2 byte:		13037	13041
2020-08-19 14:17:49,583 # 12 - transfer 1000 times 100 byte:		105926	15360
2020-08-19 14:17:49,615 # 13 - transfer 1000 times 2 byte to register:	26370	26374
2020-08-19 14:17:49,739 # 14 - transfer 1000 times 100 byte to register:119140	28618
2020-08-19 14:17:49,761 # 15 - acquire/release 1000 times:		17648	17653
2020-08-19 14:17:50,763 # -- - SUM:					852112	308867
2020-08-19 14:17:50,763 # 
2020-08-19 14:17:50,767 # ### All runs complete ###
```

**Comment by benpicco** [comment]
Reminds me that we still need some [heuristics](https://github.com/RIOT-OS/RIOT/pull/14261#issuecomment-644314287) to decide below which transfer size polling is actually faster than DMA setup.

File: boards/common/blxxxpill/include/periph_conf.h [hunk 2]
```diff
@@ -174,23 +195,35 @@ static const uart_conf_t uart_config[] = {
         .rx_pin     = GPIO_PIN(PORT_A, 10),
         .tx_pin     = GPIO_PIN(PORT_A, 9),
         .bus        = APB2,
-        .irqn       = USART1_IRQn
+        .irqn       = USART1_IRQn,
+#ifdef MODULE_PERIPH_DMA
+        .dma        = 2,
+        .dma_chan   = 2
```
Comment (bergzand) [inline boards/common/blxxxpill/include/periph_conf.h:201]: The `.dma_chan` is not used on the stm32f1 (and similar). Can you amend your commit and set all `.dma_chan` members to zero and maybe put a comment after each mentioning that the value is unused. Something like:
```suggestion
        .dma_chan   = 0 /* Unused on the stm32f1 */
```

**Review by fjmolinas — APPROVED** [review]
ACK, @hugueslarrive provided test output and the changes match the datasheet.

**Pull request merged**